packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -962,12 +961,10 @@ export class Project extends ProjectBase {
         }),
       ],
     });
-    this.role.attachInlinePolicy(policy);
-
-    // add an explicit dependency between the EC2 Policy and this Project -
+    // add an explicit dependency between the VPC Policy and this Project -
     // otherwise, creating the Project fails,
     // as it requires these permissions to be already attached to the Project's Role
-    const cfnPolicy = policy.node.findChild('Resource') as CfnResource;
+    const cfnPolicy = policy.node.defaultChild as CfnResource;
```

Comment:
Does this work in the current version? From what I can tell, `LazyPolicy` extends `Policy`, and so this will leave a dependency in the template to a resource that will be later removed, if I'm correct.